src/Microsoft.DotNet.ProjectJsonMigration/ProjectMigrator.cs

```diff
@@ -56,7 +63,19 @@ private MigrationRuleInputs ComputeMigrationRuleInputs(MigrationSettings migrati
             var propertyGroup = templateMSBuildProject.AddPropertyGroup();
             var itemGroup = templateMSBuildProject.AddItemGroup();
 
-            return new MigrationRuleInputs(projectContexts, templateMSBuildProject, itemGroup, propertyGroup);
+            return new MigrationRuleInputs(projectContexts, templateMSBuildProject, itemGroup, propertyGroup, xproj);
+        }
+
+        private string FindXprojFile(string projectDirectory)
+        {
+            var allXprojFiles = Directory.EnumerateFiles(projectDirectory, "*.xproj", SearchOption.TopDirectoryOnly);
+
+            if (allXprojFiles.Count() > 1)
+            {
+                throw new Exception("Multiple xproj files found in {projectDirectory}, please specify which to use");
```

Comment:
Why isn't this a MigrationError instead?  When do you throw an exception vs. a MigrationError.Throw()?